cub/test/c2h/generators.cu

```diff
@@ -36,12 +36,31 @@
 
 #include <c2h/custom_type.cuh>
 #include <c2h/generators.cuh>
-#include <curand.h>
 #include <fill_striped.cuh>
 
+#if C2H_HAS_CURAND 
+#include <curand.h>
+#else
+#include <thrust/random.h>
+#endif
+
 namespace c2h
 {
 
+#if !C2H_HAS_CURAND 
+struct i_to_rnd_t
+{
+  thrust::default_random_engine m_engine{};
+
+  template <typename IndexType>
+  __host__ __device__ float operator()(IndexType n)
+  {
+    m_engine.discard(n);
```

Comment:
Just leaving this here for posterity. 
I was not sure about the complexity of `discard(n)` for thrust's default re. Per this comment that @senior-zero pointed me to, https://github.com/senior-zero/thrust_benchmark/blob/f193da487300b8534550dd15f2d3e86310062249/benches/tbm/range_generator.cuh#L145-L147 we believe it's `O(log(n))`. 
Hence, we decided to dismiss this complexity concern.